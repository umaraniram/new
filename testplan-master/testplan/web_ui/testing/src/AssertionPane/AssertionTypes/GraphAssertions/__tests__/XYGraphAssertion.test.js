import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import XYGraphAssertion from '../XYGraphAssertion.js';

function defaultProps() {
  return {
    assertion: {
                "meta_type": "entry",
                "utc_time": "2019-07-12T12:36:09.782759+00:00",
                "machine_time": "2019-07-12T12:36:09.782759+00:00",
                "type": "Graph",
                "series_options": {
                                     'Data Name': {"colour": "red"}
                                  },
                "line_no": 50,
                "graph_data": {
                    "Data Name": [{
                            "x": "A",
                            "y": 10
                        }, {
                            "x": "B",
                            "y": 5
                        }, {
                            "x": "C",
                            "y": 15
                        }
                    ]
                },
                "description": "Bar Graph",
                "category": "DEFAULT",
                "graph_options": {"legend": true},
                "graph_type": "Bar"
            }
    };
}


describe('XYGraphAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('can render basic markup without error', () => {
    shallow(<XYGraphAssertion {...props}/>).html();
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<XYGraphAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });

});