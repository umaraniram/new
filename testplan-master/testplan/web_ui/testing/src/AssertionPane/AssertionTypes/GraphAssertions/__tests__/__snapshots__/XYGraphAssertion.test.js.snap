// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XYGraphAssertion shallow renders the correct HTML structure 1`] = `
<div
  className="centreComponent_1moixrt"
>
  <XYPlot
    animation={true}
    className=""
    height={500}
    width={750}
    xType="ordinal"
  >
    <HorizontalGridLines
      attr="y"
      direction="horizontal"
    />
    <XAxis
      attr="x"
      attrAxis="y"
      orientation="bottom"
    />
    <ChartLabel
      className="x-axis-label"
      includeMargin={false}
      style={
        Object {
          "textAnchor": "middle",
          "transform": "rotate(0)",
        }
      }
      text=""
      xPercent={0.5}
      yPercent={1.107}
    />
    <YAxis
      attr="y"
      attrAxis="x"
      orientation="left"
    />
    <ChartLabel
      className="y-axis-label"
      includeMargin={false}
      style={
        Object {
          "textAnchor": "middle",
          "transform": "rotate(270)",
        }
      }
      text=""
      xPercent={-0.0455}
      yPercent={0.5}
    />
    <VerticalBarSeries
      className=""
      color="red"
      data={
        Array [
          Object {
            "x": "A",
            "y": 10,
          },
          Object {
            "x": "B",
            "y": 5,
          },
          Object {
            "x": "C",
            "y": 15,
          },
        ]
      }
      key="Data Name"
      stack={false}
      style={Object {}}
    />
    <HighlightOverlay
      className=""
      color="rgb(77, 182, 172)"
      enableX={true}
      enableY={true}
      onBrushEnd={[Function]}
      onDrag={[Function]}
      opacity={0.3}
    />
  </XYPlot>
  <DiscreteColorLegendItem
    className=""
    colors={
      Array [
        "#12939A",
        "#79C7E3",
        "#1A3177",
        "#FF9833",
        "#EF5D28",
      ]
    }
    items={
      Array [
        Object {
          "color": "red",
          "title": "Data Name",
        },
      ]
    }
    orientation="horizontal"
    width={750}
  />
</div>
`;
